### PR TITLE
COSBETA-176: Index document in the after_save hook

### PR DESCRIPTION
**Pull request by kicferk1**
Add elasticsearch indexing when a notification is updated from non-complete to complete. 

## Checklist:

- [x] Automated checks are passing locally.

### CODE REVIEW

File: cosmetics-web/app/models/notification.rb [hunk 1]
```diff
@@ -24,6 +24,7 @@ class Notification < ApplicationRecord
 
   before_save :add_product_name, if: :will_save_change_to_product_name?
   before_save :add_import_country, if: :will_save_change_to_import_country?
+  after_save :update_elasticsearch_index
```
Comment (broder) [inline cosmetics-web/app/models/notification.rb]: Could / should we do this in an `aasm` callback in the `submit_notification` event? https://github.com/aasm/aasm#callbacks

Comment (kicferk1) [reply cosmetics-web/app/models/notification.rb]: We could. I was worried that if we do that, and then add a different event that transitions(from a possibly different state) to `notification_complete` this bug could come back. Is that likely to happen?

Comment (broder) [reply cosmetics-web/app/models/notification.rb]: I don't think there will be another event to go to `notification_complete` but @DWRendell or @tom-bell-softwire might know more?

Comment (tom-bell-softwire) [reply cosmetics-web/app/models/notification.rb]: I think all notifications should transition via the `submit_notification` event (even if submitting an update/correction in the future), so this is probably the best place to add the callback.

**Pull request merged**